library/private/options_tools.h

```diff
@@ -354,13 +354,55 @@ std::string format(color_t var)
 
 //----------------------------------------------------------------------------
 /**
- * Format provided var into a string from provided direction_t
- * rely on format(std::vector<double>&)
+ * Format provided var into a string from provided direction_t.
+ * Format as `+X`/`+X-Y`/... if possible, otherwise rely on `format(std::vector<double>&)`
  */
 std::string format(direction_t var)
 {
-  // TODO generate a proper direction string
-  return options_tools::format(static_cast<std::vector<double>>(var));
+  const auto is_zero = [](double a) { return std::abs(a) < 1e-12; };
+  const auto abs_diff = [](double a, double b) { return std::abs(a) - std::abs(b); };
+  const auto formatAsXYZ = [&]()
+  {
+    std::string str = "";
+    double first_nonzero = 0; // not non-zero until first
+    char sign = ' ';          // initially not `+`/`-` to force first sign
+    for (size_t i = 0; i < 3; ++i)
+    {
+      const double v = var[i];
+      if (!is_zero(v))
+      {
+        if (is_zero(first_nonzero))
+        {
+          first_nonzero = v;
+        }
+        else if (!is_zero(abs_diff(v, first_nonzero)))
+        {
+          throw std::invalid_argument("not all same");
```

Comment:
i dont get this code path, what does this correspond to ?

Reply:
we can only write as `±XYZ` if there are one or more non-zero components and they all have the same absolute value (`1,0,0`, `2,2,0`, `3,-3,3`, ...) if either condition doesn't hold then we bail an fall back to vector format